Sources/SwiftSVG/Fill.swift

```diff
@@ -2,11 +2,15 @@ import Swift2D
 
 public struct Fill {
     
-    public var color: String?
-    public var opacity: Double?
+    public var color: String
+    public var opacity: Double
     public var rule: Rule = .nonZero
     
-    public init() {}
+    public init(color: String?, opacity: Double?, rule: Rule?) {
+        self.color = color ?? "black"
+        self.opacity = opacity ?? 1
+        self.rule = rule ?? .nonZero
+    }
```

Comment:
Let's move the default values to the initializer.
```suggestion
    /// Initialize aggregate data related to a `Fill`.
    ///
    /// - parameters:
    ///   - color: Name of or hexadecimal value used to color the fill. Default: "black"
    ///   - opacity: The level of opaqueness given applied to the color. Default: 1.0
    ///   - rule: Algorithm used to determine the inside of the shape. Default: `.nonZero`
    public init(
        color: String = "black", 
        opacity: Double = 1.0, 
        rule: Rule = .nonZero
    ) {
        self.color = color
        self.opacity = opacity
        self.rule = rule
    }
```

Reply:
Oh, I forgot to mention that I didn't supply default values for the constructor to prevent the type of bug that was in your original code where you had forgotten to supply the path's fill rule to the struct's constructor. Without default values such bug is more difficult to make, as you have to supply some value.
Hence, I see various drawbacks to this kind of constructor.

Reply:
I would like to maintain a parameterless initialization option. If the properties of the `Fill` struct are going to be non-optional, then those default values need to be present somewhere. I prefer them to be as defaults specified in the method signature rather than as _un-exposed_ values with the init method. This adds in documentation efforts well for those who are not looking directly at the code.

Reply:
The parameterless initialisation is easy of course, it can simply call self.init(...) with whatever arguments it likes. I've already done this on the 'rpoelstra' branch in my fork of your code, where I've already added some other construct as well.
Fair enough about the unexposed values. Maybe define them as static constants on the struct to have them readily available instead of hardcoding them in various places?